universal-application-tool-0.0.1/app/models/StoredFile.java

```diff
@@ -0,0 +1,33 @@
+package models;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.Transient;
+import play.data.validation.Constraints;
+
+/** The ebean mapped class for a file stored in AWS S3 */
+@Entity
+@Table(name = "files")
+public class StoredFile extends BaseModel {
+  private static final long serialVersionUID = 1L;
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  @Constraints.Required String name;
+
+  public byte[] getContent() {
+    return data;
```

Comment:
*question*: I'm a little concerned about this being in-memory - is that avoidable?  Can we use streams?  Should we use streams?

Reply:
Applicant-uploaded files should never have a reason for their contents to be on the UAT server itself.

Admin-uploaded PDF templates, generated PDFs, and CSVs will have to be on the UAT when they're consumed or generated. I agree that it'd be preferable to use streams for those when feasible.

Reply:
Question for either of you: if we need to shell out for PDF generation, what kind of disk or /tmp directory behavior can we expect with our current infra setup?

Reply:
Looks like we get [fixed 20GB temp disk storage](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/fargate-task-storage.html)

Reply:
Ah cool, that's way more than enough.